clar2wasm/src/wasm_generator.rs

```diff
@@ -498,6 +498,66 @@ impl<'a> ASTVisitor<'a> for WasmGenerator {
         Ok(builder)
     }
 
+    fn traverse_comparison<'b>(
```

Comment:
I think this could be simplified to a `visit_comparison`, and just use the default traversal. When possible, we should default to changing the visitor only. It's only when we need to emit some instruction before the children that we should modify the traverser.

Reply:
Yep, absolutely.

Reply:
Thanks. I'll also create a new branch to add some documentation around this to the ASTVisitor.